### ISSUE TITLE
Infinite loop in 3.9-EA when warming up partitions

### ISSUE THREAD

**Issue by nlwillia** [hazelcast/hazelcast#11567]
I have a cluster where I want to do some dynamic config and load some data before letting migration occur.  I tried the following and got an infinite loop.

````
HazelcastInstance hz = Hazelcast.newHazelcastInstance();

hz.getCluster().changeClusterState(ClusterState.NO_MIGRATION);

hz.getConfig().addRingBufferConfig(new RingbufferConfig("test").setCapacity(1000));
````

WARNING: [192.168.198.1]:5702 [dev] [3.9-EA] Partitions can't be assigned since cluster-state= NO_MIGRATION

Here's a partial trace:
````
	  at com.hazelcast.internal.partition.impl.PartitionStateManager.isPartitionAssignmentAllowed(PartitionStateManager.java:180)
	  at com.hazelcast.internal.partition.impl.PartitionStateManager.initializePartitionAssignments(PartitionStateManager.java:131)
	  at com.hazelcast.internal.partition.impl.InternalPartitionServiceImpl.firstArrangement(InternalPartitionServiceImpl.java:244)
	  at com.hazelcast.internal.partition.impl.InternalPartitionServiceImpl.getPartitionOwner(InternalPartitionServiceImpl.java:190)
	  at com.hazelcast.internal.partition.PartitionServiceProxy$PartitionProxy.getOwner(PartitionServiceProxy.java:220)
	  at com.hazelcast.internal.util.InvocationUtil.warmUpPartitions(InvocationUtil.java:111)
	  at com.hazelcast.internal.util.InvocationUtil.invokeOnStableClusterSerial(InvocationUtil.java:68)
	  at com.hazelcast.internal.dynamicconfig.ClusterWideConfigurationService.broadcastConfig(ClusterWideConfigurationService.java:219)
	  at com.hazelcast.internal.dynamicconfig.DynamicConfigurationAwareConfig.addRingBufferConfig(DynamicConfigurationAwareConfig.java:553)
````

I guess I can configure before I change the cluster state, but the infinite loop was unexpected.

**Comment by tombujok** [comment]
@vbekiaris Could you have a look at it tomorrow?

**Comment by vbekiaris** [comment]
@nlwillia thanks for the issue report. This is no longer the case in 3.9-SNAPSHOT (fixed by #11520 ). When you try your test code with the latest snapshot release it works as expected. You can find the snapshots repositories URLs here: https://github.com/hazelcast/hazelcast#snapshot-releases

(thanks to @tkountis for the quick investigation)
